tox.ini

```diff
@@ -6,22 +6,79 @@
 ; we set the associated flag (e.g. for linting we don't need
 ; the package installation).
```

Comment:
Has this been synced with agent-academy-1?

Reply:
Yes, with a small modification: in [testenv:black] and [testenv:black-check] I stayed with click=8.0.2 instead of 8.0.4 to match open-autonomy. 

I still have some trouble with pylint: agent-academy-1 does not have `pyproject.toml`. If I run `tox -p -e pylint`in this repo without that file I get:
`ERROR: No pyproject.toml or setup.py file found.`

But if I copy it from open-aea:
```
ERROR: pyproject.toml file found.
To use a PEP 517 build-backend you are required to configure tox to use an isolated_build:
```
Now, open-autonomy does not have that config and does not show that error. Even if I add it to this tox:
`ERROR: missing build-backend key at build-system section inside /home/david/Valory/repos/dev-template/pyproject.toml`

Reply:
Ok, using setup.py fixed it.

Reply:
The only problem now (although irrelevant) is that when running make generators, the hash all command gives an error because there are no packages.